### PR TITLE
`DocGroup` in scarb-doc

### PR DESCRIPTION
**Pull request by piotmag769**

### CODE REVIEW

File: extensions/scarb-doc/Cargo.toml [hunk 1]
```diff
@@ -14,17 +14,21 @@ camino.workspace = true
...
 cairo-lang-defs.workspace = true
+cairo-lang-doc.workspace = true
+cairo-lang-filesystem.workspace = true
+cairo-lang-parser.workspace = true
 cairo-lang-semantic.workspace = true
+cairo-lang-starknet.workspace = true
 cairo-lang-syntax.workspace = true
-cairo-lang-filesystem.workspace = true
 cairo-lang-utils.workspace = true
+itertools.workspace = true
```
Comment (maciektr) [inline extensions/scarb-doc/Cargo.toml:24]: Is this used here?
Same for `camino`.

File: Cargo.toml [hunk 2]
```diff
@@ -99,6 +100,7 @@ ra_ap_toolchain = "0.0.218"
 rayon = "1.10"
 redb = "2.1.1"
 reqwest = { version = "0.11", features = ["gzip", "brotli", "deflate", "json", "stream"], default-features = false }
+salsa = "0.16.1"
```
Comment (maciektr) [inline Cargo.toml:103]: Please exclude this from `non critical` group in dependabot config

**Review by maciektr — APPROVED** [review]
LGTM

File: extensions/scarb-doc/Cargo.toml (hunk 1 above)
Comment (piotmag769) [reply extensions/scarb-doc/Cargo.toml:24]: No idea tbh, I just changed the order, will check

File: extensions/scarb-doc/src/types.rs [hunk 3]
```diff
@@ -167,7 +170,7 @@ pub struct Constant {
 }
 
 impl Constant {
-    pub fn new(db: &dyn DefsGroup, id: ConstantId, node: &ast::ItemConstant) -> Self {
+    pub fn new(db: &dyn DocGroup, id: ConstantId, node: &ast::ItemConstant) -> Self {
```
Comment (mkaput) [inline extensions/scarb-doc/src/types.rs:173]: TBH It'd be easier if you just use `&ScarbDocDatabase` directly. At least that's the conclusion I did in LS.

File: extensions/scarb-doc/src/db.rs [hunk 4]
```diff
@@ -0,0 +1,120 @@
...
+/// The Cairo compiler Salsa database tailored for scarb-doc usage.
+#[salsa::database(
+    FilesDatabase,
+    ParserDatabase,
+    SyntaxDatabase,
+    DefsDatabase,
+    SemanticDatabase,
+    DocDatabase
+)]
+pub struct ScarbDocDatabase {
```
Comment (mkaput) [inline extensions/scarb-doc/src/db.rs:28]: FYI you can do the following if you wish (no need to change)

```suggestion
/// The Cairo compiler Salsa database tailored for scarb-doc usage.
#[salsa::database(
    FilesDatabase,
    ParserDatabase,
    SyntaxDatabase,
    DefsDatabase,
    SemanticDatabase,
    cairo_lang_doc::db::DocDatabase
)]
pub struct DocDatabase {
```

File: extensions/scarb-doc/src/db.rs [hunk 5]
```diff
@@ -0,0 +1,120 @@
...
+
+impl ScarbDocDatabase {
+    pub fn new(project_config: Option<ProjectConfig>) -> Self {
+        let mut db = Self {
+            storage: Default::default(),
+        };
+
+        init_files_group(&mut db);
+
+        let plugin_suite = [get_default_plugin_suite(), starknet_plugin_suite()]
```
Comment (mkaput) [inline extensions/scarb-doc/src/db.rs:40]: consider leaving a note somewhere that scarb-provided plugins have to be supported. I'd just put `// FIXME(piotmag769): blahblah` here

File: extensions/scarb-doc/src/db.rs [hunk 6]
```diff
@@ -0,0 +1,120 @@
...
+            .into_iter()
+            .fold(PluginSuite::default(), |mut acc, suite| {
+                acc.add(suite);
+                acc
+            });
+
+        db.apply_plugin_suite(plugin_suite);
```
Comment (mkaput) [inline extensions/scarb-doc/src/db.rs:47]: You probably also want to set `#[cfg(doc)]` here

**Review by mkaput — APPROVED** [review]
Looks great! Leaving some suggestions that you can apply here or other PRs :)

File: extensions/scarb-doc/src/db.rs (hunk 5 above)
Comment (maciektr) [reply extensions/scarb-doc/src/db.rs:40]: I think this should be a whole issue not TODO 😅

Comment (mkaput) [reply extensions/scarb-doc/src/db.rs:40]: True dat

File: extensions/scarb-doc/src/types.rs (hunk 3 above)
Comment (piotmag769) [reply extensions/scarb-doc/src/types.rs:173]: I have to redo the #1315 PR completely anyways (2137 conflicts) so I will do it there

File: extensions/scarb-doc/src/db.rs (hunk 5 above)
Comment (piotmag769) [reply extensions/scarb-doc/src/db.rs:40]: https://github.com/software-mansion/scarb/issues/1412

**Pull request merged**